app/assets/javascripts/comfortable_mexican_sofa/admin/modules/EmbeddedGoogleAnalytics.js

```diff
@@ -0,0 +1,173 @@
+define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
+  'use strict';
+
+  var defaultConfig = { };
+
+  function EmbeddedGoogleAnalytics($el, config) {
+    EmbeddedGoogleAnalytics.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(EmbeddedGoogleAnalytics);
+
+  EmbeddedGoogleAnalytics.componentName = 'EmbeddedGoogleAnalytics';
+
+  EmbeddedGoogleAnalytics.prototype.init = function(initialised) {
```

Comment:
Although this method is fine, its better to separate the concerns into smaller, testable functions. Perhaps one to embed the platform.js stuff, one to bind the ready function, etc.